psycopg/psycopg/_typeinfo.py

```diff
@@ -163,14 +162,21 @@ def _get_info_query(cls, conn: "BaseConnection[Any]") -> Query:
 
     @classmethod
     def _has_to_regtype_function(cls, conn: "BaseConnection[Any]") -> bool:
-        # introduced in PostgreSQL 9.4 and CockroachDB 22.2
+        # to_regtype() introduced in PostgreSQL 9.4 and CockroachDB 22.2
         info = conn.info
-        return info.vendor == "PostgreSQL" or (
-            info.vendor == "CockroachDB" and info.server_version >= 220200
-        )
+        if info.vendor == "PostgreSQL":
+            return info.server_version >= 90400
+        elif info.vendor == "CockroachDB":
+            return info.server_version >= 220200
+        else:
+            return False
 
     @classmethod
     def _to_regtype(cls, conn: "BaseConnection[Any]") -> "SQL":
+        # `to_regtype()` returns the type oid or NULL, unless the :: operator,
```

Comment:
`unless` here seems out of place. maybe while sounds better? (not a native speaker, so feel free to ignore)

Reply:
Right, it should have been "unlike" 🙂